### PR TITLE
Removing expired OCP report data

### LINKED ISSUE #247: Remove expired OCP Data
Masu removes all expired report data (default is 3 months).  OCP report data removal needs to be implemented such that it plugs into the existing expired data removal celery task.

Acceptance Criteria:
* OCP report data with reporting period begin dates that are before the masu expiration date should be removed as part of the celery removal task.
* OCP report data should be removed when the OCP provider is removed in koku

### PR DESCRIPTION
**Pull request by dccurtis**
Closes #247.

Removing expired OCP report data in the masu celery task.  This is triggered by the expired_data test end point, the monthly celery task, and removing a provider via koku.

*Testing*
1.  Ingest OCP Data.  Hit the expired_data and verify that no report data is removed when within the expiration date window
2. Adjust masu's date so that OCP data has expired.
3.  Issue DEL /expired_data and verify that the report data is removed
4.  Re-ingest OCP data and remove OCP provider from koku and verify that data is removed in the database.

*Testing Results*
[masu_ocp_cleanup_ut.txt](https://github.com/project-koku/masu/files/2531640/masu_ocp_cleanup_ut.txt)

### CODE REVIEW

File: masu/processor/ocp/ocp_report_db_cleaner.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                qty = self._accessor.get_report_query_report_period_id(report_period_id).delete()
+                LOG.info('Removing %s usage period items for usage period id %s',
+                         qty, report_period_id)
+
+            LOG.info('Report data removed for usage period ID: %s with interval start: %s',
+                     report_period_id, removed_usage_start_period)
+            removed_items.append({'usage_period_id': report_period_id,
+                                  'interval_start': str(removed_usage_start_period)})
+
+        if simulate is False:
```
Comment (42B) [inline masu/processor/ocp/ocp_report_db_cleaner.py]:
```python
if not simulate:  # ?
```

> Don’t compare boolean values to True or False using ==:
> 
> Yes:
> 
>     if greeting:
> 
> No:
> 
>     if greeting == True:
> 
> Worse:
> 
>     if greeting is True:

* https://pep8.org/#programming-recommendations

File: masu/processor/ocp/ocp_report_db_cleaner.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Removes report data from database."""
+# pylint: skip-file
```
Comment (chambridge) [inline masu/processor/ocp/ocp_report_db_cleaner.py]: why are we skipping pylint?

File: masu/processor/expired_data_remover.py [hunk 3]
```diff
@@ -15,16 +15,21 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 """Remove expired report data."""
+# pylint: skip-file
```
Comment (chambridge) [inline masu/processor/expired_data_remover.py]: why are we skipping pylint?

Comment (dccurtis) [reply masu/processor/expired_data_remover.py]: We are doing this in a few other files due to this.  For some reason adding just the exception for duplicate-code is not working (maybe this https://github.com/PyCQA/pylint/issues/214):

```
from masu.external import (AMAZON_WEB_SERVICES,
                           AWS_LOCAL_SERVICE_PROVIDER,
                           OCP_LOCAL_SERVICE_PROVIDER,
                           OPENSHIFT_CONTAINER_PLATFORM)
from masu.external.date_accessor import DateAccessor (duplicate-code)
```

File: masu/processor/ocp/ocp_report_db_cleaner.py (hunk 2 above)
Comment (dccurtis) [reply masu/processor/ocp/ocp_report_db_cleaner.py]: meant to come back to this one.  I need to try generalize `purge_expired_report_data` for both OCP and AWS.  I can do that now.

File: masu/processor/expired_data_remover.py (hunk 3 above)
Comment (blentz) [reply masu/processor/expired_data_remover.py]: We should include a comment next to the disable to explain why it's there and when it should be removed.

**Pull request merged**